src/WidgetBase.ts

```diff
@@ -271,17 +285,42 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 			delete this._previousProperties[propertyName];
 		});
 
-		const diffPropertiesResult = this.diffProperties(this._previousProperties, properties);
-		this._properties = assign(diffPropertiesResult.properties, diffPropertyResults);
+		const diffPropertiesDecorators = this.getDecorator('diffProperties');
+		if (diffPropertiesDecorators.length) {
+			const diffDecorator = diffPropertiesDecorators[diffPropertiesDecorators.length - 1];
```

Comment:
in my head it seems the most logical at least. when i write a widget i'm essentially stamping the diff behaviour with that decorator. If someone writes a mixin with additive behaviour they can still use the granular `@diffProperty`. It does bring up a valid other point in relation to my previous PR though. What should we do in the case of a base class using `@diffProperty` and a class extending it overriding it? im not sure of the behaviour.